cli/main.rs

```diff
@@ -684,56 +682,49 @@ async fn bundle_command(
   Ok(())
 }
 
-struct DocLoader {
-  fetcher: FileFetcher,
-  maybe_import_map: Option<ImportMap>,
+/// When parsing lib.deno.d.ts, only `DocParser::parse_source` is used,
+/// which never even references the loader, so this is just a stub for that scenario.
+struct StubDocLoader;
```

Comment:
We should refactor better so that we don't have some totally useless code.  If it isn't needed, the consumer should have an optional reference to a `impl DocFileLoader` versus a "stub".

Reply:
Agree, however it requires changing APIs in `deno_doc`. I'd be fine with landing as is for 1.8 as this PR actually fixes a problem. 

@Liamolucko please add a TODO here mentioning this required change to `deno_doc`. Also it'd be great if code handling doc generation was moved to `tools/doc.rs`.